prometheus/registry.go

```diff
@@ -505,22 +470,30 @@ func (r *registry) writePB(encoder expfmt.Encoder) error {
 		sort.Sort(metricSorter(mf.Metric))
 	}
 
-	// Write out MetricFamilies sorted by their name.
+	// Write out MetricFamilies sorted by their name, skipping those without
+	// metrics.
```

Comment:
How do you expose a MetricFamily without any metrics?
The expfmt package returns an encoding error in that case. (which is how I noticed the case in the first place.)